worker/caasapplicationprovisioner/manifold.go

```diff
@@ -11,8 +11,9 @@ import (
 	"github.com/juju/worker/v3/dependency"
 
 	"github.com/juju/juju/api/base"
-	apicaasapplicationprovisioner "github.com/juju/juju/api/controller/caasapplicationprovisioner"
-	caasunitprovisionerapi "github.com/juju/juju/api/controller/caasunitprovisioner"
+	"github.com/juju/juju/api/controller/caasapplicationprovisioner"
+	"github.com/juju/juju/api/controller/caasunitprovisioner"
+	"github.com/juju/juju/api/controller/controllerapi"
```

Comment:
The trade-off of one facade per worker doesn't seem worth it to me -- for controller workers. As you say, controller workers tend to have broader permissions. In addition, it's not actually a security help, as we're in the controller process anyway, so the worker code could tamper if it wanted it. There's a lot of boilerplate required for the one-facade-per-worker model, for what real value?

My thinking is that we'll establish a pattern of each worker having a `Facade` interface (we can debate the name; I don't love the word "facade") that is defined in that worker package with the list of methods that the worker uses. That's the very simple, Go-idiomatic way of describing what methods a worker interacts with.

And as long as we follow a naming convention (eg: always calling the interface `Facade`) these methods can even be machine-auditable if we need that.

Reply:
However, I do agree about pulling these out of the "api" directory. They are a Go API, but I agree it's an overloaded term and confusing. Would love people's thoughts on where these should live / what they should be called!

Reply:
One more thing: each instance of the "controller API" would have a model ID in it (forwarded to state), so its methods could only interact with a single model -- this would prevent the most egregious errors of "oops, I stomped on the wrong model".